src/phone-input/country-select-dropdown.js

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright (c) 2018-2019 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import React from 'react';
+import {List, AutoSizer} from 'react-virtualized';
+
+import Flag from './flag.js';
+import {SIZE} from './constants.js';
+import {
+  StyledCountrySelectDropdownContainer as DefaultContainer,
+  StyledCountrySelectDropdownListItem as DefaultListItem,
+  StyledCountrySelectDropdownFlagColumn as DefaultFlagColumn,
+  StyledCountrySelectDropdownNameColumn as DefaultNameColumn,
+  StyledCountrySelectDropdownDialcodeColumn as DefaultDialcodeColumn,
+} from './styled-components.js';
+import {getOverrides} from '../helpers/overrides.js';
+
+import type {CountrySelectDropdownPropsT} from './types.js';
+
+export default function CountrySelectDropdown(
+  props: CountrySelectDropdownPropsT,
+) {
+  const {
+    children,
+    dropdownHeight = '400px',
+    mapIsoToLabel,
+    overrides = {
+      CountrySelectDropdown: {},
+      CountrySelectDropdownListItem: {},
+      CountrySelectDropdownFlagColumn: {},
+      CountrySelectDropdownNameColumn: {},
+      CountrySelectDropdownDialcodeColumn: {},
+    },
+  } = props;
+  const [Container, containerProps] = getOverrides(
+    overrides.CountrySelectDropdown,
+    DefaultContainer,
+  );
+  const [ListItem, listItemProps] = getOverrides(
+    overrides.CountrySelectDropdownListItem,
+    DefaultListItem,
+  );
+  const [FlagColumn, flagColumnProps] = getOverrides(
+    overrides.CountrySelectDropdownFlagColumn,
+    DefaultFlagColumn,
+  );
+  const [NameColumn, nameColumnProps] = getOverrides(
+    overrides.CountrySelectDropdownNameColumn,
+    DefaultNameColumn,
+  );
+  const [Dialcode, dialcodeProps] = getOverrides(
+    overrides.CountrySelectDropdownDialcodeColumn,
+    DefaultDialcodeColumn,
+  );
+  return (
+    <Container $height={dropdownHeight} {...containerProps}>
+      <AutoSizer>
+        {({height, width}) => {
+          return (
+            <List
+              role="listbox"
+              height={height}
+              width={width}
+              rowCount={children.length}
+              rowHeight={32}
+              rowRenderer={({index, key, style}) => {
+                // resetMenu and getItemLabel should not end up on native html elements
+                const {resetMenu, getItemLabel, ...rest} = children[
+                  index
+                ].props;
+                return (
+                  <ListItem
+                    key={key}
+                    style={style}
+                    {...rest}
+                    {...listItemProps}
+                  >
+                    <FlagColumn {...flagColumnProps}>
+                      <Flag
+                        iso2={children[index].props.item.id}
+                        size={SIZE.compact}
```

Comment:
is directly defining `SIZE.compact` intentional? should this change in coordination with the input size?

Reply:
does the normal select menu change in size based on top level `size` prop?

personally, i don't *think* the size of the input should change the size of the dropdown but that is purely my own intuition so should double check with design.

Reply:
that makes sense - given that the size never changes, should the various flag sizes be maintained?

Reply:
ah those are to match the input size itself:

![phone input sizes](https://user-images.githubusercontent.com/1939257/58722631-c1b47500-838c-11e9-80d7-baa5db2ba4af.png)

Reply:
a dev could use overrides to make things larger but maybe i should add a prop for sizing the dropdown holistically

Reply:
it lgtm for now, maybe we can allow further customization when needed

Reply:
They needed for the selected state I assume that would be different for different size inputs.